cli/dcoscli/cluster/main.py

```diff
@@ -321,5 +321,10 @@ def _needs_cluster_cert(dcos_url):
             requests.get(dcos_url, timeout=http.DEFAULT_TIMEOUT)
         except requests.exceptions.SSLError:
             return True
+        except Exception as e:
+            logger.warning(
+                'Unexpected exception occurred while calling %s: %s',
+                dcos_url, e)
+            return True
```

Comment:
It seems a bit strange to have `True` be the default error case, but maybe I'm missing something here.

Reply:
`_needs_cluster_cert` makes some sort of pre-setup request to check if https works out of the box, but I had miss the case where there is an even harder failure (eg. route not found or similar), in such a situation it is currently failing with a stack trace, with this change it would fail properly later on as shown by `test_setup_unreachable_url`.